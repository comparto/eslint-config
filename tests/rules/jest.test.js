const testingLibrary = require('../../src/rules/jest/testing-library')
const jestDom = require('../../src/rules/jest/jest-dom')

describe('jest', () => {
  beforeEach(() => {
    jest.resetModules()
  })
  it('should have default jest rules', () => {
    const jestRules = require('../../src/rules/jest')

    expect(jestRules).toMatchSnapshot()
  })

  it('should also include testing-library & jest-dom rules', () => {
    jest.doMock('../../src/utils', () => ({
      doesDependencyExist: () => true
    }))
    const jestRules = require('../../src/rules/jest')

    expect(jestRules.plugins).toStrictEqual([
      'jest',
      'testing-library',
      'jest-dom'
    ])
    expect(jestRules.rules).toEqual(
      expect.objectContaining(testingLibrary.rules),
      expect.objectContaining(jestDom.rules)
    )
  })
})
