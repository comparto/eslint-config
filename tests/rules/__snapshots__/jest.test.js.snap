// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest.js 1`] = `
Object {
  "env": Object {
    "jest/globals": true,
  },
  "overrides": Array [
    Object {
      "files": Array [
        "**/__tests__/**/*",
        "**/*.{spec,test}.*",
      ],
      "rules": Object {
        "max-lines-per-function": "off",
        "react/display-name": "off",
      },
    },
  ],
  "plugins": Array [
    "jest",
  ],
  "rules": Object {
    "jest/consistent-test-it": Array [
      "error",
      Object {
        "fn": "test",
        "withinDescribe": "it",
      },
    ],
    "jest/expect-expect": "warn",
    "jest/lowercase-name": "error",
    "jest/no-commented-out-tests": "error",
    "jest/no-conditional-expect": "error",
    "jest/no-deprecated-functions": "warn",
    "jest/no-disabled-tests": "warn",
    "jest/no-done-callback": "error",
    "jest/no-duplicate-hooks": "error",
    "jest/no-export": "error",
    "jest/no-focused-tests": "error",
    "jest/no-hooks": "off",
    "jest/no-identical-title": "error",
    "jest/no-if": "error",
    "jest/no-interpolation-in-snapshots": "error",
    "jest/no-jasmine-globals": "warn",
    "jest/no-jest-import": "error",
    "jest/no-large-snapshots": Array [
      "warn",
      Object {
        "inlineMaxSize": 6,
        "maxSize": 60,
      },
    ],
    "jest/no-mocks-import": "error",
    "jest/no-restricted-matchers": "off",
    "jest/no-standalone-expect": "error",
    "jest/no-test-prefixes": "error",
    "jest/no-test-return-statement": "off",
    "jest/no-try-expect": "error",
    "jest/prefer-called-with": "error",
    "jest/prefer-expect-assertions": "off",
    "jest/prefer-hooks-on-top": "error",
    "jest/prefer-spy-on": "off",
    "jest/prefer-strict-equal": "off",
    "jest/prefer-todo": "warn",
    "jest/require-to-throw-message": "off",
    "jest/require-top-level-describe": "off",
    "jest/valid-describe": "error",
    "jest/valid-expect": "error",
    "jest/valid-expect-in-promise": "error",
    "jest/valid-title": "error",
  },
}
`;
